crates/geometry/src/line_segment.rs

```diff
@@ -1,15 +1,24 @@
 use std::f32::consts::TAU;
 
 use approx::{AbsDiffEq, RelativeEq};
-use nalgebra::{vector, Point2, Vector2};
 use serde::{Deserialize, Serialize};
 
-use crate::{arc::Arc, orientation::Orientation};
+use coordinate_systems::{vector, Point2, Vector2};
 
-#[derive(Clone, Copy, Debug, Default, Deserialize, PartialEq, Serialize)]
-pub struct LineSegment(pub Point2<f32>, pub Point2<f32>);
+use crate::{arc::Arc, direction::Direction};
 
-impl AbsDiffEq for LineSegment {
+#[derive(Clone, Copy, Debug, Default, Deserialize, Serialize)]
+#[serde(bound = "")]
+pub struct LineSegment<Frame>(pub Point2<Frame>, pub Point2<Frame>);
+
+// Manual implementation required because the derived version imposes Frame to be PartialEq
```

Comment:
but it is ... Maybe you missed it when adding it?

Reply:
Right, this was before I gave up and just derived every necessary trait for all of our frames.
I had hoped that would not be necessary, but tag type parameters that don't actually appear in the struct hierarchy are sadly not supported well by derive macros.

Reply:
Replaced most manual trait implementations with derives on all Line types